### DISCUSSION TITLE
Elementor WP hook to enqueue a css file inside elementor editor

### DISCUSSION THREAD

**Discussion by r-interactive** [elementor/elementor#24789] [Developers Community]
### Start a discussion

Is there an elementor hook or a wordpress hook to enqueue a css or js file only to the Elementor editor?

I've tried adding the wp_enqueue_style() code to these hooks below, but it still doesn't load when I am in the elementor editor page.

add_action( 'after_setup_theme', 'hello_elementor_child_setup' );
add_action( 'wp_enqueue_scripts', 'hello_elementor_child_scripts_styles', 20 );

I want to hide the header + footer sections only in the elementor editor pages, but those will show normally on the front-end.

thank you!

**Comment by perks98** [comment]
To add scripts or styles in the editor only I have found these pages that might help you on their developers website: 
Script: [https://developers.elementor.com/docs/scripts-styles/editor-scripts/](https://developers.elementor.com/docs/scripts-styles/editor-scripts/)
Styles: [https://developers.elementor.com/docs/scripts-styles/editor-styles/](https://developers.elementor.com/docs/scripts-styles/editor-styles/)

**Reply by r-interactive** [reply]
Thank you @perks98 ! using add_action( 'elementor/editor/before_enqueue_styles', 'my_plugin_editor_styles' ) or 
add_action( 'elementor/editor/after_enqueue_styles', 'my_plugin_editor_styles' ) works to add custom stylesheets. 
However, it doesn't seems to hide my header section. I am not using elementor theme builder to create the header + footer, I am using header.php + footer.php php files.

My admin editor css file, I have:
#masthead{
/*header 'id' is 'masthead'*/
  display: none !important;
}

to try to hide the header in the elementor editor, but it's not working. Would you know why it's not hiding the header?

thanks!

**Reply by perks98** [reply]
@r-interactive if you are using Google Chrome or FireFox, if you inspect the header do you see your CSS styling and if so can you see other CSS that is overwriting your display value?

**Reply by r-interactive** [reply]
@perks98 - It's loading my custom css file inside the elementor editor, but the masthead doesn't seem to be taking that css 'display:none;'.  The header + footer sections are all inside the  <iframe> in the editor, is that the reason why it's not taking that css?

**Reply by perks98** [reply]
@r-interactive you are right, you can't style content inside an iframe normally. I think instead of the editor hook try the preview styles hook:
[https://developers.elementor.com/docs/scripts-styles/preview-styles/](https://developers.elementor.com/docs/scripts-styles/preview-styles/)

**Reply by r-interactive** [reply]
thank @perks98 ! Using 'elementor/preview/enqueue_styles' works! You are the best!
